### PR TITLE
Feature/new hire experience

### PR DESCRIPTION
**Pull request by RMaksymczuk**
A functionality for 'New Hire' that allows to add proper experience using a special button

image

### CODE REVIEW

File: src/app/header/header.component.html [hunk 1]
```diff
@@ -6,10 +6,10 @@
   </button>
   <div class="collapse navbar-collapse" id="headerNavbar" *ngIf="auth.user | async">
     <ul class="navbar-nav mr-auto">
-      <li class="nav-item active">
+      <li class="nav-item" [ngClass]="{'active': (router.url == '/user/dashboard')}">
```
Comment (Sobol) [inline src/app/header/header.component.html]: I'm not sure if it's too elegant to check for the current route in html. https://stackoverflow.com/a/37976498/9769449

File: src/app/project/project.component.html [hunk 2]
```diff
@@ -11,7 +11,7 @@
 <p class="description">
   <inline-editor type="textarea" [(ngModel)]="project.desc" (onSave)="saveProject('desc', $event)" cols="70" rows="5" name="customEditableText"></inline-editor>
 </p>
-<p class="technologies" *ngIf="project.technologies.length > 0">
+<p class="technologies" *ngIf="project.technologies && project.technologies.length > 0">
```
Comment (Sobol) [inline src/app/project/project.component.html]: Is this needed? Does it throw error when it's blank? Maybe we could use some sort of route resolver(if it's not yet in place)

File: src/app/user-settings/user-settings.component.html [hunk 3]
```diff
@@ -190,6 +190,7 @@ <h1 class="card-title pricing-card-title text-center custom-card-title">
           </div>
         </div>
         </ng-template>
+        <button type="button" class="btn btn-lg btn-block btn-outline-primary custom-block-button" (click)=addNewHireExperience() *ngIf="showNewHireExperienceButton()">New Hire</button>
```
Comment (Sobol) [inline src/app/user-settings/user-settings.component.html]: This should actually be checked in ngOnInit. Simple bool to indicate the actual value. This is due the fact that every method you use in view will be triggered n-times while the view is rendered. Add simple console.log("Test") to showNewHireExperienceButton and check logs to see what I'm talking about 😄

File: src/app/user-settings/user-settings.component.ts [hunk 4]
```diff
@@ -140,6 +144,25 @@ export class UserSettingsComponent implements OnInit {
     this.save(false);
   }
 
+  addNewHireExperience() {
+    let currentDate = new Date();
```
Comment (Sobol) [inline src/app/user-settings/user-settings.component.ts]: I have a feeling they can be switched to const instead let

File: src/app/user/user-experience.component.html [hunk 5]
```diff
@@ -22,7 +22,7 @@
       </app-project>
       <button class="btn btn-outline-danger btn-sm" (click)="removeMainProject(j)" *ngIf="editForm">Remove main project</button>
     </div>
-    <p *ngIf="experience.projects" class="working-on">
+    <p *ngIf="experience.projects && experience.projects.length > 0" class="working-on">
```
Comment (Sobol) [inline src/app/user/user-experience.component.html]: The same case as in src/app/project/project.component.html.

File: src/app/project/project.component.html (hunk 2 above)
Comment (RMaksymczuk) [reply src/app/project/project.component.html]: You're right, it's not needed, as I remember i've got some other problem that was triggering the error message, but in the end, the double check in this place is not necessary.

File: src/app/user-settings/user-settings.component.html (hunk 3 above)
Comment (RMaksymczuk) [reply src/app/user-settings/user-settings.component.html]: Good point, calling a function inside *ngIf is not a good solution. But I can't call it only on ngOnInit. I need to check it also after "New Hire" button click and after any experience removing action. Will fix it.

File: src/app/header/header.component.html (hunk 1 above)
Comment (RMaksymczuk) [reply src/app/header/header.component.html]: Sure, will check the link.

**Pull request merged**